examples/heterogeneous-node-classification-with-hashgnn.ipynb

```diff
@@ -0,0 +1,534 @@
+    "But instead of doing neural transformations like most GNNs, its transformations are done by locality sensitive min-hashing.\n",
+    "\n",
+    "We will give hashGNN the `plot_keywords` node properties as input, and it will output new feature vectors for each node that has been enriched by message passing over relationships.\n",
+    "Since the `plot_keywords` vectors are already binary we don't have to do any [binarization](https://neo4j.com/docs/graph-data-science/current/machine-learning/node-embeddings/hashgnn/#_feature_binarization_2) of the input.\n",
+    "\n",
+    "Since we have multiple node labels and relationships, we make sure to enable the heterogeneous capabilities of HashGNN by setting `heterogeneous=True`.\n",
+    "Notably we also declare that we want to include all kinds of nodes, not only the `Movie` nodes we will train on, by explicitly specifying the `contextNodeLabels`.\n",
+    "\n",
+    "Please see the [HashGNN documentation](https://neo4j.com/docs/graph-data-science/current/machine-learning/node-embeddings/hashgnn/) for more on this algorithm."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5e254efe",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add a HashGNN node property step to the pipeline\n",
+    "_ = pipe.addNodeProperty(\n",
+    "    \"beta.hashgnn\",\n",
+    "    mutateProperty=\"embedding\",\n",
+    "    iterations=4,\n",
+    "    heterogeneous=True,\n",
+    "    embeddingDensity=512,\n",
+    "    neighborInfluence=0.8,\n",
+    "    featureProperties=[\"plot_keywords\"],\n",
+    "    randomSeed=41,\n",
+    "    contextNodeLabels=G.node_labels(),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "30ff6713",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set the embeddings vectors produced by HashGNN as feature input to our ML algorithm\n",
+    "_ = pipe.selectFeatures(\"embedding\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6f4f2bc1",
+   "metadata": {},
+   "source": [
+    "## Setting up autotuning\n",
+    "\n",
+    "It is time to set up the [ML algorithms](https://neo4j.com/docs/graph-data-science/current/machine-learning/training-methods/) for the training part of the pipeline.\n",
+    "\n",
+    "In this example we will add logistic regression and random forest algorithms as candidates for the final model.\n",
+    "Each candidate will be evaluated by the pipeline, and the best one, according to our specified metric, will be chosen.\n",
+    "\n",
+    "It is hard to know how much regularization we need so as not to overfit our models on the training dataset, and for this reason we will use the autotuning capabilities of GDS to help us out.\n",
+    "The autotuning algorithm will try out several values for the regularization parameters `penalty` (of logistic regression) and `minSplitSize` (of random forest) and choose the best ones it finds.\n",
```

Comment:
could optionally mention that its tuned on a logarithmic scale which explains that it makes sense to put things like (1e-x, 1ey) as range and the value of x matters even if for example 1e-3 and 1e-6 are close in an absolute sense.

perhaps its too much detail

Reply:
interesting idea, I'll see if I can work it in without making it too complicated

Reply:
decided against, to keep it simple